tests/test_class2D.py

```diff
@@ -0,0 +1,330 @@
+        self.assertTrue(rmse < utest_tolerance(self.dtype))
+
+    def testComplexConversionErrors(self):
+        """
+        Test we raise when passed incorrect dtypes.
+
+        Also checks we can handle 0d vector in `to_real`.
+
+        Most other cases covered by classification unit tests.
+        """
+
+        with pytest.raises(
+            TypeError, match="coef provided to to_complex should be real."
+        ):
+            _ = self.fspca_basis.to_complex(
+                np.arange(self.fspca_basis.count, dtype=np.complex64)
+            )
+
+        with pytest.raises(
+            TypeError, match="coef provided to to_real should be complex."
+        ):
+            _ = self.fspca_basis.to_real(
+                np.arange(self.fspca_basis.count, dtype=np.float32).flatten()
+            )
+
+    def testRotate(self):
+        """
+        Trivial test of rotation in FSPCA Basis.
+
+        Also covers to_real and to_complex conversions in FSPCA Basis.
+        """
+        coef = self.fspca_basis.expand_from_image_basis(self.imgs)
+        # rotate by pi
+        rot_coef = self.fspca_basis.rotate(coef, radians=np.pi)
+        rot_imgs = self.fspca_basis.evaluate_to_image_basis(rot_coef)
+
+        for i, img in enumerate(self.imgs):
+            rmse = np.sqrt(np.mean(np.square(np.flip(img) - rot_imgs[i])))
+            self.assertTrue(rmse < 10 * utest_tolerance(self.dtype))
+
+
+class RIRClass2DTestCase(TestCase):
+    def setUp(self):
+        self.resolution = 16
+        self.dtype = np.float64
+
+        # Create some projections
+        v = Volume(
+            np.load(os.path.join(DATA_DIR, "clean70SRibosome_vol.npy")).astype(
+                self.dtype
+            )
+        )
+        v = v.downsample(self.resolution)
+
+        # Clean
+        self.clean_src = Simulation(
+            L=self.resolution,
+            n=321,
+            vols=v,
+            dtype=self.dtype,
+        )
+
+        # With Noise
+        noise_var = 0.01 * np.var(np.sum(v[0], axis=0))
+        noise_filter = ScalarFilter(dim=2, value=noise_var)
+        self.noisy_src = Simulation(
+            L=self.resolution,
+            n=123,
+            vols=v,
+            dtype=self.dtype,
+            noise_filter=noise_filter,
+        )
+
+        # Set up FFB
+        # Setup a Basis
+        self.basis = FFBBasis2D((self.resolution, self.resolution), dtype=self.dtype)
+        # Calculate Fourier Bessel Coefs
+        self.clean_coefs = self.basis.evaluate_t(
+            self.clean_src.images(0, self.clean_src.n)
+        )
+        self.noisy_coefs = self.basis.evaluate_t(
+            self.noisy_src.images(0, self.clean_src.n)
+        )
+
+        # Create Basis
+        self.clean_fspca_basis = FSPCABasis(
+            self.clean_src, self.basis, noise_var=0
+        )  # Note noise_var assigned zero, skips eigval filtering.
+        self.clean_fspca_basis.build(self.clean_coefs)
```

Comment:
This computes the coefs by default if not supplied, correct? May be good to exercise that behavior.

Reply:
Right, I added that later and was only using in notebooks.  Good catch. Converted one of the existing test lines to cover this case.